### DISCUSSION TITLE
Python DB API 2.0 Driver and SQLAlchemy Dialect for QuestDB & Superset Integration

### DISCUSSION THREAD

**Discussion by srinify** [questdb/questdb#1877] [Ideas]
Hello QuestDB community! My name's Srini and I'm involved with the Apache Superset (https://github.com/apache/superset) project. Superset was originally designed to work well with Apache Druid and has since expanded to support nearly any SQL speaking database. 

Superset especially shines in time-series visualization, which is why QuestDB came to mind. I'd love to advocate for the creation of a a Python DB API 2.0 driver + SQLAlchemy dialect (as a single library) for QuestDB so Quest and Superset can start to jam together.

Besides this driver, there is a tiny amount of integration work needed on the Superset side. I wrote about that here: https://preset.io/blog/building-database-connector/

**Comment by puzpuzpuz** [comment]
Many thanks for starting this thread. We're certainly aware of Apache Superset and know that some of our users would love to see the QuestDB dialect in SQLAlchemy. We do have plans to improve SQLAlchemy and Apache Superset support, but it's hard to name a concrete milestone for that since right now we have a bunch of highly demanded features on the roadmap.

Needless to say that if a community member contributes the QuestDB dialect in SQLAlchemy, that would be just awesome.

**Comment by pswu11** [comment]
A relevant issue created by the community some time ago https://github.com/questdb/questdb/issues/1028
